### DISCUSSION TITLE
基于阿里云oss做相应压力测试，性能极差

### DISCUSSION THREAD

**Discussion by yanruogu** [juicedata/juicefs#1296] [General]
我这边参考阿里云nas的测试方式对juicefs做了性能测试： https://help.aliyun.com/document_detail/95501.html?spm=5176.21213303.J_6704733920.7.500c3edacyrZui&scm=20140722.S_help%40%40%E6%96%87%E6%A1%A3%40%4095501._.ID_help%40%40%E6%96%87%E6%A1%A3%40%4095501-RL_nas%20%E6%B5%8B%E8%AF%95-OR_main-V_2-P0_0

这边使用的对象存储是阿里云oss，元数据存储为redis： 
```
docker run -d --net host redis
```
通过如下指令创建了存储： 
```
juicefs format     --storage oss     --bucket https://douyu-juicefs     --access-key xxx    --secret-key xxx redis://xxx:6379/1     douyu-juicefs
```
执行挂载： 
```
/usr/bin/juicefs mount -d redis://xxx:6379/1 /mnt
```
在实际测试时，有点儿不太敢相信自己的眼睛，这性能也太差了： 
![image](https://user-images.githubusercontent.com/5188514/149447845-76fc1b55-fd66-43b0-8099-2ca5a96a1e50.png)
![image](https://user-images.githubusercontent.com/5188514/149447868-2df825b4-a0c4-4840-80d6-0e6eb4f71dff.png)
![image](https://user-images.githubusercontent.com/5188514/149447887-cdb5130e-e8f4-4e2a-a237-6537d044b881.png)

是阿里oss性能就这么差，还是我哪里操作有问题？

**Comment by davies** [comment]
以上分别测试的是 4K 随机读 4K 随机写和 1M 随机读，一下分别看下：

1. 4K随机读在没有缓存的情况下，取决于 OSS 的随机读的延迟，大概在 20-50ms，对应的 IOPS 是 20-50，测试过程中会逐步建立缓存，IOPS 也会逐步提高。取决于你使用的版本，默认缓存大小可能是 1GB，导致无法被完整缓存，IOPS 会一直上不去。
2. 4K 随机写时，会先缓存到内存中再异步并发写入到 OSS，看到 IOPS 会高很多。
3. 1MB 随机读跟 4K 类似，在没有缓存时取决于 OSS的随机读延迟，大概在 50-100ms, 对应的 IOPS 是 10-20, 

JuiceFS 支持随机读写的场景，因为对象存储的延迟很高，更擅长的是顺序读写，需要根据你实际的应用场景测试对应的读写模式，请看 https://juicefs.com/docs/zh/community/performance_evaluation_guide
